### PR TITLE
[extended-monitoring][image-availability-exporter] Refactor TLS interactions

### LINKED ISSUE #262: Support custom CAs in the image-availability-exporter
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Some custom registries' certificates may be signed by a custom CA.

### Proposed Solution

Consider providing a custom CA to image-availiabity-exporter.

1. We've got a custom `ca` field in the `registry-secret`.
2. We've got to enhance `image-availability-exporter` so that it supports custom CAs

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by asviel**
## Description

Support custom container registry CA, registry credentials and insecure (HTTP) registries in the image-availability-exporter. Change ModuleConfig settings.

## Why do we need it, and what problem does it solve?

Closes #262

## Why do we need it in the patch release (if we do)?

We don't.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: extended-monitoring
type: feature
summary: Support custom container registry CA, registry credentials and insecure (HTTP) registries in the image-availability-exporter. Change ModuleConfig settings.
impact_level: default
```

### CODE REVIEW

**Review by MariaMakeeva — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by asviel** [comment]
@z9r5, @MariaMakeeva please, help me with descriptions in docs

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @asviel. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/340-extended-monitoring/openapi/doc-ru-config-values.yaml [hunk 1]
```diff
@@ -6,12 +6,26 @@ properties:
...
           Включен ли imageAvailabilityExporter.
-      skipRegistryCertVerification:
-        description: |
-          Пропускать ли валидацию TLS-сертификата для Docker Registry.
       ignoredImages:
         description: |
           Список образов, для которых нужно пропустить проверку, например `alpine:3.12` или `quay.io/test/test:v1.1`.
+      registry:
+        description: |
+          <PLACEHOLDER>
```
Comment (z9r5) [inline modules/340-extended-monitoring/openapi/doc-ru-config-values.yaml:14]:
```suggestion
          Настройки подключения к container registry.
```

File: modules/340-extended-monitoring/openapi/config-values.yaml [hunk 2]
```diff
@@ -24,6 +19,33 @@ properties:
...
+          scheme:
+            type: string
+            description: |
+              Container registry access scheme.
+            enum: [HTTP, HTTPS]
+            default: HTTPS
+          tlsConfig:
+            type: object
+            description: |
+              <PLACEHOLDER>
```
Comment (z9r5) [inline modules/340-extended-monitoring/openapi/config-values.yaml:37]:
```suggestion
              Connection settings for container registry.
```

**Review by z9r5 — APPROVED** [review]

File: modules/340-extended-monitoring/openapi/doc-ru-config-values.yaml [hunk 3]
```diff
@@ -6,12 +6,26 @@ properties:
...
+      registry:
+        description: |
+          Настройки подключения к container registry.
+        properties:
+          scheme:
+            description: |
+              Протокол доступа к container registry.
+          tlsConfig:
+            description: |
+              <PLACEHOLDER>
```
Comment (z9r5) [inline modules/340-extended-monitoring/openapi/doc-ru-config-values.yaml:21]:
```suggestion
              Настройки подключения к container registry
```

File: modules/340-extended-monitoring/openapi/config-values.yaml [hunk 4]
```diff
@@ -24,6 +19,33 @@ properties:
           A list of images to ignore when checking the presence in the registry, e.g., `alpine:3.12` or `quay.io/test/test:v1.1`.
         items:
           type: string
+      registry:
+        type: object
+        description: |
+          <PLACEHOLDER>
```
Comment (z9r5) [inline modules/340-extended-monitoring/openapi/config-values.yaml:25]:
```suggestion
          Connection settings for container registry.
```

**Review by nabokihms — APPROVED** [review]

**Pull request merged**